### PR TITLE
Components: Set type=button for TabPanel button elements.

### LINKED ISSUE #11767: Components: Tab Panel: Butttons trigger form submission
**Describe the bug**
The buttons in the tab panel component trigger form submission.

**To Reproduce**
1. Include a TabPanel within a `<form>` element with an `onSubmit` handler.
2. Click on one of the tab panel buttons. 
3. The form's `onSubmit` handler is triggered.

**Expected behavior**
The tab to change but the form not to be submitted.

**Possible Solutions**
1. `preventDefault` on the click event.
2. Add a `type="button"` to the `TabButton` component.
3. Allow passing additional props to the `TabButton` via the `tabs` prop.

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Safari, Chrome, etc...
 - Version 12

### PR DESCRIPTION
**Pull request by TimothyBJacobs**
Fixes #11767.

## Description
The default button type is "submit" which submits the enclosing form.
Setting the type property to "button" prevents this behavior.

I don't think it ever makes sense that you'd want a button in a tab panel to submit the form,
so I just made the `type="button"` behavior for everything. Otherwise, a prop could be added to adjust the button type. 

## How has this been tested?
I tried to write a unit test for this, but apparently an `onSubmit` handler on a form will never be called from a button being clicked because it's not actually simulating the DOM?

## Types of changes
Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Review by ajitbohra — APPROVED** [review]
LGTM :+1:

**Comment by aduth** [comment]
Had you considered changing this to use the [`Button` component](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/button), which defaults to `type="button"`?

**Comment by TimothyBJacobs** [comment]
I hadn’t. That would make sense though. I’ll see what that looks like.

**Comment by TimothyBJacobs** [comment]
Switching out for the `<Button>` component looks good.

**Comment by aduth** [comment]
There's a stylistic difference:

Before|After
---|---
screen shot 2018-12-11 at 2 50 01 pm|screen shot 2018-12-11 at 2 51 00 pm

I'm not really sure what this component is meant to look like. It seems a bit useless in its current form.

**Comment by TimothyBJacobs** [comment]
Ah. I guess because I already had provided the styles for my usage of it.

Yeah, style wise the existing component is incredibly barebones.

**Comment by gziolo** [comment]
@jasmussen would mind giving it a sping to validate it doesn't introduce any regressions for tabs?

**Comment by jasmussen** [comment]
@gziolo Thanks for the ping. I took this for a spin and added this to the separator to test:

tabpanel

It doesn't seem like we're using this component anywhere in Gutenberg itself. It really seems like it should come with some basic styles, and I'm essentially echoing @aduth here. This is because the `Button` component unsets the `appearance` and `border` of the `button`, so it's really quite bare.

But maybe that shouldn't hold up this PR if it fixes an issue? Not sure, and not sure how I can review this from a design perspective as there doesn't seem to be a design.

To put it more simply, if the above was the design we intended for the TabPanel component, it would not be shippable. But this could be a separate task, to improve that, as mentioned.

**Comment by gziolo** [comment]
I merged this PR as is given that it doesn't have any impact on Gutenberg itself as I confirmed that this component is never used in the codebase. The change itself fixes the issue it references but let's work on the styles related improvements in a separate issue. I opened #13587 to track it. Let's discuss further steps there as it needs some design involvement.
